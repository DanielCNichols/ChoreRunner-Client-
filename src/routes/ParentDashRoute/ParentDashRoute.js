import React, { Component, useState, useContext, useEffect } from 'react';
import ApiService from '../../services/api-service.js';
import '../../components/ParentDashboard/ParentDashboard.css';
import AddHouseHoldForm from '../../components/AddHouseHoldForm/AddHouseHoldForm';
import s from './ParentDashRoute.module.css';

import HouseCard from '../../components/HouseCard/HouseCard';

export default function ParentDashRoute(props) {
  const [households, setHouseholds] = useState([]);
  const [houseAdd, setHouseAdd] = useState(false);
  const [memberAdd, setMemberAdd] = useState(false);
  const [editHouse, setEditHouse] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    ApiService.getHouseholds()
      .then(res => {
        setHouseholds(res);
      })
      .catch(error => {
        setError(error);
      });
  }, []);

  const UserPrompt = () => {
    return (
      <>
        <div className={s.parentDashboardFeedback}>
          <h3>Get Started!</h3>

          <p>1. Click the "Add Group" button to create groups</p>
          <p>2. Add members to your groups with the "Add Member" button</p>
          <p>
            3. Manage your group's tasks by clicking <span>'Manage tasks'</span>{' '}
          </p>
        </div>
      </>
    );
  };

  const toggleAddHouse = () => {
    setHouseAdd(!houseAdd);
  };

  const toggleEditHouse = () => {
    setEditHouse(!editHouse);
  };
  //handleAdd
  const handleAddHousehold = newHousehold => {
    setHouseholds([newHousehold, ...households]);
  };

  //handleDelete
  const handleDeleteHousehold = id => {
    let updated = households.filter(house => {
      return house.id !== id;
    });

    setHouseholds(updated);
  };

  //HandleEdit
  const handleEditHousehold = updated => {
    let idx = households.findIndex(house => house.id === updated.id);

    let newHouseholds = [...households];

    newHouseholds[idx].name = updated.name;
    setHouseholds(newHouseholds);
  };

  const handleAddMembers = member => {
    let index = households.findIndex(house => (house.id = member.household_id));

    let newHouseholds = [...households];

    newHouseholds[index].members.push(member);
    setHouseholds(newHouseholds);
  };

  const deleteMember = (memberId, householdId) => {
    let household = households.findIndex(house => house.id === householdId);

    let newHouseholds = [...households];

    let newMembers = newHouseholds[household].members.filter(
      member => member.id !== memberId
    );

    newHouseholds[household].members = newMembers;

    setHouseholds(newHouseholds);
  };

  return (
    <section className={s.parentDashboard}>
      <header>
        <h3>Your Groups</h3>
      </header>

      {/* !I think we can refactor this and just put the add member button in the card! */}
      <div className={s.addControls}>
        <button className="arcadeButton" onClick={() => toggleAddHouse()}>
          Add Group
        </button>
      </div>

      <div className={s.formContainer}>
        {houseAdd && (
          <AddHouseHoldForm
            handleCancel={toggleAddHouse}
            handleAdd={handleAddHousehold}
            toggleAdd={toggleAddHouse}
          />
        )}
      </div>

      {!households.length && !houseAdd ? (
        <UserPrompt />
      ) : (
        households.map(house => {
          return (
            <HouseCard
              key={house.id}
              house={house}
              editing={editHouse}
              toggleEdit={toggleEditHouse}
              handleEdit={handleEditHousehold}
              handleDelete={handleDeleteHousehold}
              handleAddMembers={handleAddMembers}
              deleteMember={deleteMember}
            />
          );
        })
      )}
    </section>
  );
}
