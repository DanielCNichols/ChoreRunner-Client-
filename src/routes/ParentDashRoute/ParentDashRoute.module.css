@import '../../vars.css';

.parentDashboard {
  align-items: center;
  width: 90%;
  margin: 0 auto;
  height: 100%;
  padding-top: 15vh;
  padding-bottom: 200px;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
}

.addControls {
  display: none;
}

.formContainer {
  width: 100%;
}

.addControls button {
  width: 100%;
  height: 45px;
  text-align: center;
  margin: 10px auto;
  font-family: 'Press Start 2P', cursive;
  font-size: 1em;
  padding: 7px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
}

.parentDashboardFeedback h3 {
  text-align: center;
  width: 100%;
  color: white;
  font-size: 1.1em;
}

.parentDashboardFeedback {
  width: 85%;
  color: white;
  font-family: var(--font-body);
  text-align: left;
  margin: 20px auto;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 0px 10px 30px 10px;
  background: rgb(0, 0, 0, 0.5);
  position: relative;
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
  border-radius: 10px;
}

.parentDashboardFeedback ol {
  display: inline-block;
  text-align: left;
  padding: 0;
  margin: 0;
}

.parentDashboardFeedback li {
  margin: 20px 0;
}

@media screen and (min-width: 450px) {
  .formContainer {
    width: 50%;
    max-width: 500px;
    min-width: 270px;
  }

  .parentDashBoardFeedback {
    width: 500px;
  }
}

@media screen and (min-width: 800px) {
  .parentDashboard {
    width: 80%;
  }

  .formContainer {
    width: 500px;
  }

  .addControls {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 250px;
    padding: 0 20px;
  }
}
@keyframes bounce {
  from {
    transform: translateY(0px);
  }
  to {
    transform: translateY(-3px);
  }
}

@-webkit-keyframes bounce {
  from {
    transform: translateY(0px);
  }
  to {
    transform: translateY(-3px);
  }
}

@keyframes point {
  from {
    transform: translateX(0px);
  }
  to {
    transform: translateX(3px);
  }
}
@-webkit-keyframes point {
  from {
    transform: translateX(0px);
  }
  to {
    transform: translateX(3px);
  }
}
