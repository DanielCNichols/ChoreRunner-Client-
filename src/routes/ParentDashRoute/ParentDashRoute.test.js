import React from 'react';
import ReactDOM from 'react-dom';
import ParentDashRoute from './ParentDashRoute';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('ParentDashRoute component testing', () => {
  const match = {
    params: {
      id: 1,
    },
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ParentDashRoute />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ParentDashRoute />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
