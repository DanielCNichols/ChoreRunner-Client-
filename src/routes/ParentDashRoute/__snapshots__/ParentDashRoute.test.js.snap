// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentDashRoute component testing renders UI as expected 1`] = `
<section
  className="parentDashboard"
>
  <header>
    <h3
      className="videoGameTitles"
    >
      Your Groups
    </h3>
  </header>
  <div
    className="addControls"
  >
    <button
      className="arcadeButton"
      onClick={[Function]}
    >
      Add Group
    </button>
  </div>
  <div
    className="buttonMenuContainer"
  >
    <button
      className="FloatingButton floatingButton"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M16 8A8 8 0 110 8a8 8 0 0116 0zM8.5 4a.5.5 0 00-1 0v3.5H4a.5.5 0 000 1h3.5V12a.5.5 0 001 0V8.5H12a.5.5 0 000-1H8.5V4z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    className="formContainer"
  />
  <div
    className="parentDashboardFeedback"
  >
    <h3
      className="videoGameTitles"
    >
      Get Started!
    </h3>
    <ol>
      <li>
        1. Click the "Add" button to create groups
      </li>
      <li>
        2. Add members to your groups with the "Add Member" button
      </li>
      <li>
        3. Manage your group's tasks by clicking "Manage Tasks"
      </li>
    </ol>
  </div>
</section>
`;
